src/components/admin/branchSettings/BranchFormModal.tsx

```diff
@@ -146,6 +146,13 @@ const BranchFormModal: React.FC<Props> = props => {
                   <option value="STAFF">Staff</option>
                 </Select>
               </FormControl>
+              <FormControl isRequired>
+                <FormLabel>Secret</FormLabel>
+                <Select {...register("secret")} placeholder="Select option">
+                  <option value="true">Yes</option>
+                  <option value="false">No</option>
+                </Select>
+              </FormControl>
```

Comment:
Can you have this show only on application branches? Also, I think instead of a dropdown, a checkbox would be the easiest, similar to how we have grading enabled.